nomt/src/beatree/ops/update/leaf_stage.rs

```diff
@@ -22,7 +22,9 @@ use super::{
 };
 
 struct ExtendRangeResponse {
-    changed: Vec<(Key, ChangedLeafEntry)>,
+    deleted: Vec<(Key, ChangedLeafEntry)>,
+    new_base: Option<(Key, LeafNode, Option<Key>)>,
+    new_high_range: Option<Key>,
```

Comment:
just having `changed` and `new_high` would have all the information necessary to proceed.

There are 3 possibilities:
  1. The next leaf is in `changed`.
  2. The next leaf is not in changed, but a higher leaf is in `changed`. therefore, the leaf hasn't changed
  3. The next leaf is not in changed, and there is no higher leaf in `changed`. without `new_high`, this previously led to an ambiguous situation that was the source of a bug.

It would be better to have this contain the minimal information possible.

Reply:
Yes, initially, I didn't have a new_base field, but it was possibly inside the last item of `changed`. I moved it to `deleted` and `new_base` just to create less messy code when dealing with `LeavesTrackerEntry`. I'm not a fan of that enum, and as you said in another comment, we could get rid of it by extracting the range from the changed items.

Therefore, to avoid refactoring everything in this PR, I will keep this now and work on a simplification follow-up!